taskchampion/lib/src/kv.rs

```diff
@@ -100,6 +120,16 @@ impl CList for TCKVList {
     }
 }
 
+impl Default for TCKVList {
+    fn default() -> Self {
+        Self {
+            len: libc::size_t::default(),
+            _capacity: libc::size_t::default(),
+            items: null_mut(),
+        }
```

Comment:
This is one of the really ugly corner-cases for Rust / C/C++ interop: Rust doesn't necessarily represent an empty vector as (0, 0, NULL), and uses a different semaphore value in the pointer field. I think a better approach here would be something like `vec_into_raw_parts(Vec::default())`. This _might_ be the cause of the free errors you're seeing.

Reply:
https://github.com/rust-lang/rust/issues/39625 seems to be the best info I can find on this topic (it's also been discussed quite a lot internally at work, which is why it comes to mind)

Reply:
Well that wasn't it. Turns out this implementation wasn't even being used. I deleted it.